src/App.tsx

```diff
@@ -13,6 +15,22 @@ const App = () => {
     }
   }, []);
 
+  const wallet = useWallet();
+
+  /* eslint-disable @typescript-eslint/no-unsafe-call */
+  /* eslint-disable @typescript-eslint/no-unsafe-member-access */
+  window.ethereum?.on('accountsChanged', function () {
+    if (!wallet.loading) {
+      void wallet.connect('metamask');
```

Comment:
can we actually connect with different wallet other than metamask? Why do we hardcode metamask here? Please explain 🙏

Reply:
`window.ethereum?.on` listens to metamask events only. So it means a change happened while connected to metamask. We use WalletConnect as well but the WalletConnect flow is not affected.